bftclient/src/matcher.cpp

```diff
@@ -0,0 +1,61 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License"). You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "matcher.h"
+
+namespace bft::client {
+
+std::optional<Match> Matcher::onReply(UnmatchedReply&& reply) {
+  if (!valid(reply)) return std::nullopt;
+
+  auto key = MatchKey{std::move(reply.metadata), std::move(reply.data)};
+  const auto [it, success] = matches_[key].insert_or_assign(reply.rsi.from, std::move(reply.rsi.data));
+  (void)it;  // unused variable hack needed by GCC
```

Comment:
`std::tie` + `std::ignore` may be used as alternative.

Reply:
I'm sticking to the basics for now :) GCC needs to get realize the value of destructuring and ignore this :)